File: MicroTube.Client.Angular/src/app/data/DTO/VideoSearchResultDTO.ts
import { VideoDto, VideoRawDto } from "./VideoDto";

export class VideoSearchResultDto
{
  videos: VideoDto[];
  meta: string | null;
  constructor(raw: VideoSearchResultRawDto)
  {
    this.videos = raw.videos.map(_ => new VideoDto(_));
    this.meta = raw.meta;
  }
}
export interface VideoSearchResultRawDto
{
  videos: VideoRawDto[];
  meta: string | null;
}

File: MicroTube/Controllers/Videos/DTO/VideoSearchResultDTO.cs
namespace MicroTube.Controllers.Videos.DTO
{
	public class VideoSearchResultDTO
	{
		public IEnumerable<VideoDTO> Videos { get; set; }
		public string? Meta { get; set; }
		public VideoSearchResultDTO(IEnumerable<VideoDTO> videos)
		{
			Videos = videos;
		}
	}
}
